plugins/module_utils/nae.py

```diff
@@ -690,32 +691,44 @@ def create_pre_change_from_file(self):
                 self.params['cmap'] = {}
                 data = self.load(open(self.params.get('file')))
                 tree = self.construct_tree(data)
-                if tree is False:
-                    self.module.fail_json(
-                        msg="Error parsing input file, unsupported object found in hierarchy.",
-                        **self.result)
-                tree_roots = self.find_tree_roots(tree)
-                ansible_ds = {}
-                for root in tree_roots:
-                    exp = self.export_tree(root)
-                    for key, val in exp.items():
-                        ansible_ds[key] = val
-                self.copy_children(ansible_ds)
-                toplevel = {"totalCount": "1", "imdata": []}
-                toplevel['imdata'].append(ansible_ds)
-                with open(self.params.get('file'), 'w') as f:
-                    json.dump(toplevel, f)
-                del self.params['cmap']
-                f.close()
+                self.create_structured_data(tree)
             else:
                 self.module.fail_json(
                     msg="Error parsing input file. JSON format necessary",
                     **self.result)
+        else:
+            try:
+                with open(self.params.get('file')) as j:
+                    extract_data = json.load(j)
+                tree = self.construct_tree([extract_data[0]])
+                self.params['cmap'] = {}
+                self.create_structured_data(tree)
+            except KeyError:
+                pass
```

Comment:
Is that necessary?
The original code prior to this change is supposed to work as expected . That is, if **self.is_json(f.read()) is not False** the code should have executed with no errors.

What I implemented was just a work around to make the original code work as it is supposed to and gives no new functionality.
What do you think?

Reply:
When we try to extract data above that's already in JSON format: 
**{"fvTenant": {"attributes": {"name": "AnsibleTest", "dn": "uni/tn-AnsibleTest", "status": "deleted"}}}**
 we come across a KeyError. This is because we are unable to retrieve an element by using the **[extract_data[0]]** in the code. Hence, nothing is done in that case and we let the code run as is.